### PR TITLE
Add compute pipeline

### PR DESCRIPTION
**Pull request by Nastormo**

### CODE REVIEW

File: include/magma/vk/ComputePipeline.h [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    const VkPipelineLayoutCreateInfo &getPipelineLayoutInfo() const { return _pipelineLayoutInfo; }
+
+    void setLayout(const std::vector<VkDescriptorSetLayout> &descriptorSetLayouts);
+    void setLayout(const VkDescriptorSetLayout &descriptorSetLayout);
+private:
+    VkPipelineLayoutCreateInfo _pipelineLayoutInfo;
+};
+
+/**
+ * @brief Class for create compute pipeline
```
Comment (MoonDweller29) [inline include/magma/vk/ComputePipeline.h:29]: Rename to:
```
@brief Class for creating compute pipeline
```

File: include/magma/vk/ComputePipeline.h [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+ * @version 0.1
+ * @date 2021-02-09
+ */
+#pragma once
+#include <vulkan/vulkan.h>
+
+#include "shaderModule.h"
+
+/**
+ * @brief Class for  create and contain info about compute pipeline
```
Comment (MoonDweller29) [inline include/magma/vk/ComputePipeline.h:13]: Maybe that'll be better:
```
@brief Class that contains info for creating compute pipeline
```

File: src/magma/vk/ComputePipeline.cpp [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+ComputePipelineInfo::ComputePipelineInfo() {
+    _pipelineLayoutInfo = {};
+    _pipelineLayoutInfo.sType = VK_STRUCTURE_TYPE_PIPELINE_LAYOUT_CREATE_INFO;
+    _pipelineLayoutInfo.setLayoutCount = 0;
+    _pipelineLayoutInfo.pSetLayouts = nullptr;
+    _pipelineLayoutInfo.pushConstantRangeCount = 0;
+    _pipelineLayoutInfo.pPushConstantRanges = nullptr;
+}
+
+void ComputePipelineInfo::setLayout(const std::vector<VkDescriptorSetLayout> &descriptorSetLayouts) {
```
Comment (MoonDweller29) [inline src/magma/vk/ComputePipeline.cpp:20]: In GraphicsPipeline this method is called ```setLayouts``` (with 's' at the end).
So there were no overloaded methods.

File: include/magma/vk/ComputePipeline.h [hunk 4]
```diff
@@ -0,0 +1,45 @@
+/**
+ * @file ComputePipeline.h
+ * @brief Class create compute pipeline
```
Comment (MoonDweller29) [inline include/magma/vk/ComputePipeline.h:3]: There should be the same brief as in ComputePipeline

File: src/magma/vk/ComputePipeline.cpp (hunk 3 above)
Comment (Nastormo) [reply src/magma/vk/ComputePipeline.cpp]: resolve after discussion

**Review by MoonDweller29 — APPROVED** [review]

**Pull request merged**